crex/models/requirement.py

```diff
@@ -0,0 +1,19 @@
+from __future__ import annotations
+from dataclasses import dataclass
+from dataclasses_json import dataclass_json
+from frex.models import DomainObject
+from crex.models import CourseCodeRestriction
+from typing import Tuple
+from rdflib import URIRef
+
+
+@dataclass_json
+@dataclass(frozen=True)
+class Requirement(DomainObject):
+    # TODO: Requirements probably will change drastically.
```

Comment:
unclear how much more requirements will change. current modeling is now moderately flexible... i think